include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1075 @@
+                MPI_INT,
+                y_sizes.data(),
+                1,
+                MPI_INT,
+                0,
+                commGather
+            ));
+            MPI_CHECK(MPI_Gather(
+                &gatherRank,
+                1,
+                MPI_INT,
+                gatherRanks.data(),
+                1,
+                MPI_INT,
+                0,
+                commGather
+            ));
+
+            auto gatherRanks2 = gatherRanks;
+
+            sort_by_key( gatherRanks, y_offsets );
+            sort_by_key( gatherRanks2, y_sizes );
+
+
+            std::vector< int > recvcounts( gatherSize, 1 );
+
+            MPI_CHECK(MPI_Gatherv(
+                gSumMom2->getHostBuffer( ).getBasePointer( ),
+                subGrid.getLocalDomain().size.y(),
+                MPI_DOUBLE,
+                globalSumMom2.getDataPointer( ),
+                y_sizes.data(),
+                y_offsets.data(),
+                MPI_DOUBLE,
+                0,
+                commGather
+            ));
+            MPI_CHECK(MPI_Gatherv(
+                gSumPos2->getHostBuffer( ).getBasePointer( ),
+                subGrid.getLocalDomain().size.y(),
+                MPI_DOUBLE,
+                globalSumPos2.getDataPointer( ),
+                y_sizes.data(),
+                y_offsets.data(),
+                MPI_DOUBLE,
+                0,
+                commGather
+            ));
+            MPI_CHECK(MPI_Gatherv(
+                gSumMomPos->getHostBuffer( ).getBasePointer( ),
+                subGrid.getLocalDomain().size.y(),
+                MPI_DOUBLE,
+                globalSumMomPos.getDataPointer( ),
+                y_sizes.data(),
+                y_offsets.data(),
+                MPI_DOUBLE,
+                0,
+                commGather
+            ));
+            MPI_CHECK(MPI_Gatherv(
+                gCount_e->getHostBuffer( ).getBasePointer( ),
+                subGrid.getLocalDomain().size.y(),
+                MPI_DOUBLE,
+                globalCount_e.getDataPointer( ),
+                y_sizes.data(),
+                y_offsets.data(),
+                MPI_DOUBLE,
+                0,
+                commGather
+            ));
+
+            /* print timestep, emittance to file: */
+            if( writeToFile )
+            {
+                using dbl = std::numeric_limits< float_64 >;
+                outFile.precision( dbl::digits10 );
+                if (currentStep > 0.0){
+                    int startWindow_y = movingWindow.globalDimensions.offset.y();
+                    int endWindow_y = movingWindow.globalDimensions.size.y() + startWindow_y;
+                    if (fisttimestep == true)
+                    {
+                        outFile << "#step emit_all" << std::scientific;
+                        for (int i = startWindow_y; i < (endWindow_y + 10); i+=10)
+                        {
+                            outFile << " " << i * UNIT_LENGTH * SI::CELL_WIDTH_SI;
```

Comment:
it should be CELL_HEIGHT_SI and not CELL_WIDTH_SI

Reply:
@kossag14 Why do you multiply by UNIT_LENGTH ?